Failures for new `pytest-fail-slow` check in Windows CI jobs

Issue to keep track of spurious failures after gh-20672 was merged yesterday. 

This one showed up on gh-17924:
```
 ______________ TestDifferentialEvolutionSolver.test_strategy_fn _______________
[gw0] win32 -- Python 3.12.3 C:\hostedtoolcache\windows\Python\3.12.3\x64\python.exe
Test passed but took too long to run: Duration 1.9977539000001343s > 1.0s
```

I think it's just new. That test didn't even register in other recent CI runs (e.g. gh-20805), but gh-20757 merged 9 hours ago, but now in gh-20807:

================================== FAILURES ===================================
______________ TestDifferentialEvolutionSolver.test_strategy_fn _______________
[gw0] win32 -- Python 3.12.3 C:\hostedtoolcache\windows\Python\3.12.3\x64\python.exe
Test passed but took too long to run: Duration 1.9525856999998723s > 1.0s

On M3 ARM mac I see this one pretty often after bumping to latest OS, 15.2 I think:

```
================================================== FAILURES ===================================================
_____________________________________________ test_quad_vec_pool ______________________________________________
[gw7] darwin -- Python 3.12.3 /Users/treddy/python_venvs/py_312_scipy_dev/bin/python
Test passed but took too long to run: Duration 10.467066458979389s > 10s
=========================================== short test summary info ===========================================
FAILED scipy/integrate/tests/test__quad_vec.py::test_quad_vec_pool
=================== 1 failed, 58698 passed, 3734 skipped, 193 xfailed, 12 xpassed in 58.94s ===================
```

--- a/.github/workflows/linux.yml
+++ b/.github/workflows/linux.yml
@@ -233,7 +233,7 @@ jobs:
   #################################################################################
   gcc9:
     # Purpose is to examine builds with oldest-supported gcc and test with pydata/sparse.
-    name: Oldest GCC & pydata/sparse, fast, py3.11/npMin, pip+pytest
+    name: Oldest GCC & pydata/sparse, full, py3.11/npMin, pip+pytest
     needs: get_commit_message
     if: >
       needs.get_commit_message.outputs.message == 1
@@ -574,7 +574,7 @@ jobs:
 
   #################################################################################
   test_aarch64:
-    name: aarch64, fast, py3.12/npAny, pip+pytest
+    name: aarch64, fast, fail slow, py3.12/npAny, pip+pytest
     needs: get_commit_message
     if: >
       needs.get_commit_message.outputs.message == 1
@@ -598,6 +598,10 @@ jobs:
     - name: Install Python packages
       run: |
         python -m pip install -r requirements/build.txt -r requirements/test.txt
+        # We want to check for test timing only in a single job, on Linux, running the
+        # fast test suite. This is that job. See gh-20806 for previous issues
+        # after running this job on Windows and in multiple jobs.
+        python -m pip install pytest-fail-slow
 
     - name: Install SciPy
       run: |
@@ -607,4 +611,4 @@ jobs:
       run: |
         export OMP_NUM_THREADS=2
         cd ..
-        pytest --pyargs scipy -m 'not slow'
+        pytest --pyargs scipy -m 'not slow' --durations=0 --durations-min=0.5 --fail-slow=1.0

--- a/.github/workflows/linux_intel_oneAPI.yml
+++ b/.github/workflows/linux_intel_oneAPI.yml
@@ -87,7 +87,7 @@ jobs:
         shell: bash -l {0}
         run: |
           conda activate scipy-dev
-          conda install -c conda-forge pkg-config meson meson-python ninja numpy cython pybind11 pytest pytest-xdist pytest-timeout pytest-fail-slow pooch rich-click click doit pydevtool hypothesis
+          conda install -c conda-forge pkg-config meson meson-python ninja numpy cython pybind11 pytest pytest-xdist pytest-timeout pooch rich-click click doit pydevtool hypothesis
 
       - name: Initialise Intel oneAPI and Build SciPy
         shell: bash -l {0}

--- a/.github/workflows/windows.yml
+++ b/.github/workflows/windows.yml
@@ -21,8 +21,8 @@ jobs:
     name: Get commit message
     uses: ./.github/workflows/commit_message.yml
 
-  fast_dev_py_fail_slow:
-    name: fail slow, fast, py3.12/npAny, dev.py
+  fast_dev_py:
+    name: fast, py3.12/npAny, dev.py
     needs: get_commit_message
     # Ensure (a) this doesn't run on forks by default, and
     #        (b) it does run with Act locally (`github` doesn't exist there)
@@ -49,7 +49,7 @@ jobs:
 
       - name: pip-packages
         run: |
-          pip install numpy cython pybind11 pythran meson ninja pytest pytest-xdist pytest-timeout pytest-fail-slow pooch rich_click click doit pydevtool hypothesis
+          pip install numpy cython pybind11 pythran meson ninja pytest pytest-xdist pytest-timeout pooch rich_click click doit pydevtool hypothesis
           python -m pip install -r requirements/openblas.txt
 
       - name: Build
@@ -60,12 +60,12 @@ jobs:
         run: |
           # test runner parallel clashes with OpenBLAS multithreading
           $env:OPENBLAS_NUM_THREADS=1
-          python dev.py test -j2 -- --durations=0 --durations-min=0.25 --fail-slow=1.0
+          python dev.py test -j2 -- --durations=25
 
 
   #############################################################################
-  full_dev_py_min_numpy_fail_slow:
-    name: fail slow, full, py3.11/npMin, dev.py
+  full_dev_py_min_numpy:
+    name: full, py3.11/npMin, dev.py
     needs: get_commit_message
     if: >
       needs.get_commit_message.outputs.message == 1
@@ -91,7 +91,7 @@ jobs:
       - name: pip-packages
         run: |
           # 1.25.2 is currently our oldest supported NumPy version
-          python -m pip install numpy==1.25.2 cython pybind11 pythran meson-python meson ninja pytest pytest-xdist pytest-timeout pytest-fail-slow pooch rich_click click doit pydevtool hypothesis
+          python -m pip install numpy==1.25.2 cython pybind11 pythran meson-python meson ninja pytest pytest-xdist pytest-timeout pooch rich_click click doit pydevtool hypothesis
           python -m pip install -r requirements/openblas.txt
 
       - name: Build
@@ -102,7 +102,7 @@ jobs:
         run: |
           # test runner parallel clashes with OpenBLAS multithreading
           $env:OPENBLAS_NUM_THREADS=1
-          python dev.py test -j2 --mode full -- --durations=0 --durations-min=1.0 --timeout=60 --fail-slow=5.0
+          python dev.py test -j2 --mode full -- --durations=25 --timeout=60
 
 
   #############################################################################

--- a/.github/workflows/windows_intel_oneAPI.yml
+++ b/.github/workflows/windows_intel_oneAPI.yml
@@ -85,7 +85,7 @@ jobs:
       - name: Install packages from conda
         shell: cmd /C call {0}
         run: |
-          conda install -c conda-forge pkg-config meson meson-python ninja openblas libblas=*=*openblas numpy==2.0 cython pybind11 pytest pytest-xdist pytest-timeout pytest-fail-slow pooch rich-click click doit pydevtool hypothesis
+          conda install -c conda-forge pkg-config meson meson-python ninja openblas libblas=*=*openblas numpy==2.0 cython pybind11 pytest pytest-xdist pytest-timeout pooch rich-click click doit pydevtool hypothesis
 
       # MSVC is unable to compile Pythran code, therefore we need to use
       # -C-Duse-pythran=false while building SciPy.

--- a/scipy/sparse/tests/test_sparsetools.py
+++ b/scipy/sparse/tests/test_sparsetools.py
@@ -120,6 +120,7 @@ class TestInt32Overflow:
     def teardown_method(self):
         gc.collect()
 
+    @pytest.mark.fail_slow(2)  # keep in fast set, only non-slow test
     def test_coo_todense(self):
         # Check *_todense routines (cf. gh-2179)
         #

--- a/scipy/special/tests/test_mpmath.py
+++ b/scipy/special/tests/test_mpmath.py
@@ -321,6 +321,7 @@ def test_lpmv():
 # beta
 # ------------------------------------------------------------------------------
 
+@pytest.mark.slow
 @check_version(mpmath, '0.15')
 def test_beta():
     np.random.seed(1234)

--- a/scipy/stats/tests/test_stats.py
+++ b/scipy/stats/tests/test_stats.py
@@ -651,6 +651,7 @@ class TestPearsonr:
             assert_equal(res2.statistic, res.statistic)
             assert_equal(res2.pvalue, res.pvalue)
 
+    @pytest.mark.slow
     @pytest.mark.parametrize('alternative', ('less', 'greater', 'two-sided'))
     def test_bootstrap_ci(self, alternative):
         rng = np.random.default_rng(2462935790378923)

